File: backend/app/artifact_roles.py
from __future__ import annotations

from pathlib import Path


NARRATIVE_BRIEF = "narrative_brief"
SUPPORTING_DOC = "supporting_doc"
RESULTS_TABLE = "results_table"
BACKGROUND_REFERENCE = "background_reference"

ARTIFACT_ROLES = {
    NARRATIVE_BRIEF,
    SUPPORTING_DOC,
    RESULTS_TABLE,
    BACKGROUND_REFERENCE,
}


def normalize_artifact_role(value: str | None) -> str:
    role = str(value or "").strip().lower()
    if role in ARTIFACT_ROLES:
        return role
    return SUPPORTING_DOC


def infer_artifact_role(filename: str, *, requested_role: str | None = None) -> str:
    normalized = str(requested_role or "").strip()
    if normalized:
        return normalize_artifact_role(normalized)

    suffix = Path(filename).suffix.lower()
    if suffix in {".csv", ".tsv", ".xlsx", ".xls"}:
        return RESULTS_TABLE
    if suffix in {".md", ".txt", ".doc", ".docx"}:
        return SUPPORTING_DOC
    return SUPPORTING_DOC

File: frontend/src/lib/artifactRoles.ts
export const ARTIFACT_ROLE_OPTIONS = [
  { value: "narrative_brief", label: "Narrative brief" },
  { value: "supporting_doc", label: "Supporting doc" },
  { value: "results_table", label: "Results table" },
  { value: "background_reference", label: "Background / reference" },
] as const;

export type ArtifactRole = (typeof ARTIFACT_ROLE_OPTIONS)[number]["value"];

export function artifactRoleLabel(role: string) {
  return ARTIFACT_ROLE_OPTIONS.find((option) => option.value === role)?.label || "Supporting doc";
}

export function defaultArtifactRoleForFilename(filename: string): ArtifactRole {
  const suffix = filename.toLowerCase().split(".").pop() || "";
  if (["csv", "tsv", "xlsx", "xls"].includes(suffix)) {
    return "results_table";
  }
  if (["md", "txt", "doc", "docx"].includes(suffix)) {
    return "supporting_doc";
  }
  return "supporting_doc";
}
